### ISSUE TITLE
npm packages are not matching from GHSA sources

### ISSUE THREAD

**Issue by alfredodeza** [anchore/anchore-engine#633]
**Version of Anchore Engine and Anchore CLI if applicable**: current tip of master, and I believe this has been a problem for the GHSA source since it was introduced

**What happened**:

```python
            elif self.pkg_type in ['npm', 'js']:
                likematch = '%js%'
                do_langscan = True
```

Will not match against the namespace in GHSA, which is `"github:npm"`. Other languages might be affected as well, so all need to be double-checked (e.g. Ruby)

Tangentially, the `Fixed In` reporting is not entirely accurate as many affected versions can have many "fixed in" values:

```
Vulnerability ID           Package         Severity        Fix          CVE Refs             Vulnerability URL                                        Type        Feed Group        Package Path                         
GHSA-pp7h-53gx-mx7r        bl-2.2.0        High            1.2.3        CVE-2020-8244        https://github.com/advisories/GHSA-pp7h-53gx-mx7r        npm         github:npm        /node_modules/bl/package.json   
```

The above does not represent what is displayed here https://github.com/advisories/GHSA-pp7h-53gx-mx7r

**Comment by zhill** [comment]
The fix version issue looks like a problem with multiple fix records present. The current code doesn't expect more than one for a given package name, so it assumes the first match is the correct fix. I'll look into that issue further

**Comment by alfredodeza** [comment]
Looking at the relationship between the affected version and the ranges, this is how it looks when the driver parses it:

```json
   "FixedIn": [
    {
     "ecosystem": "npm",
     "identifier": "3.0.1",
     "name": "bl",
     "namespace": "github:npm",
     "range": "= 3.0.0"
    },
    {
     "ecosystem": "npm",
     "identifier": "4.0.3",
     "name": "bl",
     "namespace": "github:npm",
     "range": ">= 4.0.0 < 4.0.3"
    },
```

So it _does_ keep an identifier (version patched) with a vulnerable range.

**Comment by zhill** [comment]
Fixed by #640 . Closing.

**Comment by ScottChapman** [comment]
Thanks for the quick attention! What's the path for it to get to a released version of the engine (how can we track that)?
